### DISCUSSION TITLE
Advancing state ........

### DISCUSSION THREAD

**Discussion by Rahim-Habibi** [idaholab/moose#27500] [Q&A Modules: General] [answered]
Dear all, 

I am modeling thermo-hydro-mechanical response of a faulted geothermal reservoir. At the beginning , i  let the model to reach qausi static equilibrium but after one time step, "Advancing state " shows up and it is going on without stopping. 
here is the terminal log: 

![state](https://github.com/idaholab/moose/assets/57484835/62c33d12-8672-4610-a607-0b4217754a50)

    Building SemiLocalElemMap.                                                           [ 13.64 s] [  821 MB]
  Finished Performing Initial Setup                                                      [ 38.62 s] [  821 MB]

Time Step 0, time = -172800

Postprocessor Values:
+----------------+----------------+----------------+----------------+-------------------+
| time           | dt             | flood_count    | max_dP         | max_stress_excess |
+----------------+----------------+----------------+----------------+-------------------+
|  -1.728000e+05 |   0.000000e+00 |   0.000000e+00 |   0.000000e+00 |      0.000000e+00 |
+----------------+----------------+----------------+----------------+-------------------+

Time Step 1, time = -86400, dt = 86400

    Computing Initial Residual
      Finished Computing Residual                                                        [  5.63 s] [  867 MB]
    Finished Computing Initial Residual                                                  [  5.63 s] [  867 MB]

Performing automatic scaling calculation

    Finished Computing Residual                                                          [  5.64 s] [  868 MB]
 0 Nonlinear |R| = 6.969293e+01
    Computing Jacobian                                                                   [  7.05 s] [  876 MB]
    Finished Computing Residual                                                          [  5.65 s] [ 1097 MB]
 1 Nonlinear |R| = 2.770463e-01
    Computing Jacobian                                                                   [  7.03 s] [ 1098 MB]
    Finished Computing Residual                                                          [  5.64 s] [ 1098 MB]
 2 Nonlinear |R| = 4.924099e-03
    Computing Jacobian                                                                   [  7.03 s] [ 1098 MB]
    Finished Computing Residual                                                          [  5.64 s] [ 1098 MB]
 3 Nonlinear |R| = 9.083398e-05
    Computing Jacobian                                                                   [  7.03 s] [ 1098 MB]
    Finished Computing Residual                                                          [  5.64 s] [ 1099 MB]
 4 Nonlinear |R| = 1.944313e-06
  Finished Solving                                                                       [104.50 s] [  906 MB]
 Solve Converged!

Outlier Variable Residual Norms:
  disp_z: 1.426742e-06

Postprocessor Values:
+----------------+----------------+----------------+----------------+-------------------+
| time           | dt             | flood_count    | max_dP         | max_stress_excess |
+----------------+----------------+----------------+----------------+-------------------+
|  -1.728000e+05 |   0.000000e+00 |   0.000000e+00 |   0.000000e+00 |      0.000000e+00 |
|  -8.640000e+04 |   8.640000e+04 |   1.000000e+00 |   1.886381e+03 |     -5.240521e+06 |
+----------------+----------------+----------------+----------------+-------------------+

  Advancing State.........................................................................................

Thanks in advance for your help, 
Rahim

**Comment by milljm** [comment]
If possible, please refrain from posting pictures of text. It prevents others from searching for things like "Advancing State" and landing on posts which may be solved.

> [!NOTE]
> [Posting Guidlines](https://github.com/idaholab/moose/discussions/18270)

**Reply by Rahim-Habibi** [reply]
Thanks, I just editted.
@GiudGiud what could be the problem? it seems the issue is related to advancing the state of the FE problem by copying old solutions, shifting material properties and updating stateful material properties for the next time step etc, right? but why it stucks there? what could be the reason?
Thanks, 
Rahim

**Comment by Rahim-Habibi** [comment, marked as answer]
it seems that it was stucking in this loop in "FEProblemBase.C". 
[ 6121](https://mooseframework.inl.gov/docs/doxygen/moose/classFEProblemBase.html#a6df2c8dddd15061dd664264142e95c8e) [FEProblemBase::advanceState](https://mooseframework.inl.gov/docs/doxygen/moose/classFEProblemBase.html#a6df2c8dddd15061dd664264142e95c8e)()
 6122 {
 6123   TIME_SECTION("advanceState", 5, "Advancing State");
 6124 
 6125   for (auto & nl : [_nl](https://mooseframework.inl.gov/docs/doxygen/moose/classFEProblemBase.html#ac78551b3f3a517024f0b445ed01c9c15))
 6126     nl->copyOldSolutions();
 6127   [_aux](https://mooseframework.inl.gov/docs/doxygen/moose/classFEProblemBase.html#a021762fbebac3a426984b8ea7945cf0c)->copyOldSolutions();

i thought probably it is being stucked  in transfer the data between the time steps for auxvariable, so i removed some auxvaraibles now it runs successfully.

**Reply by GiudGiud** [reply]
how many aux variables did you have?

**Reply by Rahim-Habibi** [reply]
@GiudGiud 
I had 17, then i removed 4 of them.
